confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSourceUtil.java

```diff
@@ -215,11 +215,14 @@ private static Map<String, Object> getConsumerParams(ConfluentStreamingSourceCon
     // Spark saves the offsets in checkpoints, no need for Kafka to save them
     kafkaParams.put(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, "https");
     kafkaParams.put(CommonClientConfigs.RETRY_BACKOFF_MS_CONFIG, "500");
-    kafkaParams.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SASL_SSL");
-    kafkaParams.put(SaslConfigs.SASL_MECHANISM, "PLAIN");
-    kafkaParams.put(SaslConfigs.SASL_JAAS_CONFIG, "org.apache.kafka.common.security.plain.PlainLoginModule required " +
-      "username=\"" + conf.getClusterApiKey() + "\" password=\"" + conf.getClusterApiSecret() + "\";");
 
+    if (!Strings.isNullOrEmpty(conf.getClusterApiKey()) || !Strings.isNullOrEmpty(conf.getClusterApiSecret())) {
```

Comment:
Does validation work with just api key and not secret? If both are required, use `&&` . Consider adding validation if only one value is present. Can we also update the tooltips for these fields to indicate that they are required with confluent cloud?

Reply:
No, both are required. Changed to `&&`
Added validation if only one value is present. 
Also updated the tooltip.
22d6525bd191735661c4b0d8f51208b14365000f